BlueprintUI/Sources/Layout/LayoutWriter.swift

```diff
@@ -0,0 +1,221 @@
+        }
+    }
+    
+    public func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        nil
+    }
+}
+
+
+extension LayoutWriter {
+    
+    /// The builder is the primary surface area you interact with when using a `LayoutWriter`.
+    ///
+    /// It provides you the ability to manage the sizing and measurement of the final layout,
+    /// alongside methods to add and manage the children of the layout.
+    public struct Builder {
+        
+        //
+        // MARK: Managing Sizing
+        //
+        
+        /// How the size of the layout should be calculated. Defaults to `.unionOfChildren`,
+        /// which means the size will be big enough to contain the frames of all contained children.
+        public var sizing : Sizing = .unionOfChildren
+        
+        //
+        // MARK: Managing Children
+        //
+        
+        /// The children of the custom layout, which specifies the child element and its frame.
+        ///
+        /// Note
+        /// ----
+        /// You rarely need to access this property directly. Instead, add children via
+        /// the various provided `add(...)` methods. However, if you're `map`-ing over an array
+        /// or other collection of content, using this property directly is useful.
+        ///
+        public var children : [Child] = []
+        
+        /// Adds a new child element to the layout with the provided frame.
+        public mutating func add(
+            with frame: CGRect,
+            child : Element
+        ) {
+            self.children.append(.init(frame: frame, element: child))
+        }
+        
+        /// Adds a new child element to the layout with the provided frame.
+        /// The frame is passed to the child provider.
+        public mutating func add(
+            with frame: CGRect,
+            child : (CGRect) -> Element
+        ) {
+            self.add(with: frame, child: child(frame))
+        }
+        
+        /// Adds a new child element to the layout with the provided frame.
+        /// The frame is passed to the child provider.
+        public mutating func add(
+            with frame: () -> CGRect,
+            child : (CGRect) -> Element
+        ) {
+            let frame = frame()
+            self.add(with: frame, child: child(frame))
+        }
+        
+        /// Adds a new child element to the layout.
+        /// Using this method is helpful if you need to calculate both the frame and the element content in a single pass.
+        public mutating func add(_ child : () -> (CGRect, Element)) {
+            let result = child()
+            self.add(with: result.0, child: result.1)
+        }
+        
+        /// Enumerates each of the children, allowing you to modify them in place,
+        /// eg to align them all along a common alignment axis or to set a uniform size.
+        public mutating func modifyEach(using change : (inout Child) -> ()) {
```

Comment:
nit: is `mutate` a more typical name for this?

Reply:
Not sure honestly! Is `mutate` a common Swift thing? (I honestly dont know)

Reply:
Maybe this feeling is just from the `mutating` keyword, I can't say I have a lot of prior art to point to; no strong opinion on this!

Reply:
I do kind of prefer `modifyEach` since it matches `forEach` as well, fwiw – just `modify` or `mutate` sounds like its for one callback, not N callbacks where N is the number of children.